### PR TITLE
Describe new partial methods

### LINKED ISSUE #23084: Semantics of partial methods aren't accurately described.
C# 9.0 changed the semantics of partial methods, this page needs to be updated accordingly. New semantics: https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/extending-partial-methods.md

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d94b01ab-fe87-23fc-44b6-da9f3f627bd5
* Version Independent ID: a86008b1-2fdd-bbb9-3bbf-8fc8c51c1b71
* Content: [Classes - C# language specification](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#methods)
* Content Source: [spec/classes.md](https://github.com/dotnet/csharplang/blob/master/spec/classes.md)
* Product: **dotnet-csharp**
* Technology: **csharp-spec**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #23084

Much of this had been updated earlier this year, but this PR improves it.

### CODE REVIEW

File: docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md [hunk 1]
```diff
@@ -17,10 +17,10 @@ It is possible to split the definition of a [class](../../language-reference/key
 There are several situations when splitting a class definition is desirable:
 
 - When working on large projects, spreading a class over separate files enables multiple programmers to work on it at the same time.
-
 - When working with automatically generated source, code can be added to the class without having to recreate the source file. Visual Studio uses this approach when it creates Windows Forms, Web service wrapper code, and so on. You can create code that uses these classes without having to modify the file created by Visual Studio.
+- When using [source generators](../../roslyn-sdk/source-generator-overview.md) to generate additional functionality in a class.
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md:21]:
```suggestion
- When using [source generators](../../roslyn-sdk/source-generators-overview.md) to generate additional functionality in a class.
```

File: docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 title: "Partial Classes and Methods - C# Programming Guide"
 description: Partial classes and methods in C# split the definition of a class, a struct, an interface, or a method over two or more source files.
-ms.date: 03/23/2021
+ms.date: 06/21-2021
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md:4]:
```suggestion
ms.date: 06/21/2021
```

Comment (BillWagner) [reply docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md]: Thanks, already saw this one.

File: docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md]: and here...

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

**Pull request merged**